qiskit_experiments/framework/composite/batch_experiment.py

```diff
@@ -69,17 +86,28 @@ def circuits(self):
                 qubit_mapping = None
             else:
                 qubit_mapping = [self._qubit_map[qubit] for qubit in expr.physical_qubits]
-            for circuit in expr.circuits():
-                # Update metadata
-                circuit.metadata = {
-                    "experiment_type": self._type,
-                    "composite_metadata": [circuit.metadata],
-                    "composite_index": [index],
-                }
-                # Remap qubits if required
-                if qubit_mapping:
-                    circuit = self._remap_qubits(circuit, qubit_mapping)
-                batch_circuits.append(circuit)
+
+            if isinstance(expr, BatchExperiment):
+                circ_groups = expr._batch_circuits()
+            else:
+                circ_groups = [(expr.circuits(), expr)]
+
+            for group in circ_groups:
+                local_circuits = []
+                for circuit in group[0]:
+                    # Update metadata
+                    circuit.metadata = {
+                        "experiment_type": self._type,
+                        "composite_metadata": [circuit.metadata],
+                        "composite_index": [index],
+                    }
+                    # Remap qubits if required
+                    if qubit_mapping:
+                        circuit = self._remap_qubits(circuit, qubit_mapping)
```

Comment:
I don't think remapping logic is necessary because basically batch should be identical to
```
def batch(*exps):
    batch_circs = []
    for exp in exps:
        batch_circs.extend(transpile(exp.circuits(), self.backend, **exp.transpile_options.__dict__))
    return batch_circs
```
Actually this remap logic regenerate quantum circuit instance and this is heavy overhead without any gain. Seems like you just need to override circuit.name with `batch_` prefix.